### PR TITLE
chore: add CodeRabbit configuration file

### PR DESCRIPTION
**Pull request by Muneerali199**
Adds the standardized CodeRabbit configuration file (.coderabbit.yaml)
as discussed here :https://discord.com/channels/1022871757289422898/1235508903450775622/1474646668183081010

### CODE REVIEW

**Comment by Muneerali199** [comment]
please sir @M4dhav review it when you are free and let me know if further changes require

File: .coderabbit.yaml [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+      - label: documentation
+        instructions: Documentation updates, additions, corrections, or clarifications needed. This includes missing docs, outdated information, unclear instructions, API documentation, code examples, README improvements, and any requests for better explanations or guides.
+  planning:
+    enabled: true
+    auto_planning:
+      enabled: true
+      labels:
+        - "plan-me" # Auto-plan issues with this label
+        - "feature" # Also auto-plan these
+        - "!no-plan" # Never auto-plan issues with this label
```
Comment (M4dhav) [inline .coderabbit.yaml:31]: Turn off Planning. AI should not be used for Architecture and Feature Planning in it's entirety.

File: .coderabbit.yaml [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+    enabled: true
+    # Skip auto-review if PR title contains these keywords
+    ignore_title_keywords:
+      - "WIP"
+    # Don't auto-review draft PRs
+    drafts: false
+    # Only auto-review PRs targeting these branches
+    base_branches:
+      - main
+      - develop
```
Comment (M4dhav) [inline .coderabbit.yaml]: We use the dev branch, not develop

File: .coderabbit.yaml [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+
+  # Include a high-level summary at the start of each review
+  high_level_summary: true
+
+  # Generate sequence diagrams for complex code flows
+  sequence_diagrams: true
+
+  # Include poems in reviews
+  poem: true
```
Comment (M4dhav) [inline .coderabbit.yaml]: We don't need poems, can set this to false.

File: .coderabbit.yaml [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+
+  # Show review completion status
+  review_status: true
+
+  # Keep the walkthrough section expanded by default
+  collapse_walkthrough: false
```
Comment (M4dhav) [inline .coderabbit.yaml]: Change to true

File: .coderabbit.yaml [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+  changed_files_summary: true
+
+  # Automatically request changes on the PR (just leave comments)
+  request_changes_workflow: true
+
+  # Pre-merge checks to enforce before merging PRs
+  pre_merge_checks:
+    description:
+      # Validate that PR has a proper description
+      mode: warning # Options: off, warning, error
```
Comment (M4dhav) [inline .coderabbit.yaml]: Change to error

File: .coderabbit.yaml [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+      mode: off
+
+  # Exclude these paths from reviews (build artifacts and dependencies)
+  path_filters:
+    - "!**/node_modules/**" # npm dependencies
+    - "!**/android/**" # Native Android build files
+    - "!**/ios/**" # Native iOS build files
+    - "!**/.expo/**" # Expo build cache
+    - "!**/.expo-shared/**" # Expo shared config
+    - "!**/dist/**" # Build output
```
Comment (M4dhav) [inline .coderabbit.yaml]: These path filters are unrelated to our repository

File: .coderabbit.yaml [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+
+  # Use the following tools when reviewing
+  tools:
+    shellcheck:
+      enabled: true
+    ruff:
+      enabled: true
```
Comment (M4dhav) [inline .coderabbit.yaml]: We don't need ruff here

File: .coderabbit.yaml [hunk 8]
```diff
@@ -0,0 +1,280 @@
...
+    hadolint:
+      enabled: true
+    swiftlint:
+      enabled: true
+    phpstan:
+      enabled: true
+      level: default
+    golangci-lint:
+      enabled: true
```
Comment (M4dhav) [inline .coderabbit.yaml]: We don't need these

File: .coderabbit.yaml [hunk 9]
```diff
@@ -0,0 +1,280 @@
...
+    yamllint:
+      enabled: true
+    gitleaks:
+      enabled: true
+    checkov:
+      enabled: true
+    detekt:
+      enabled: true
```
Comment (M4dhav) [inline .coderabbit.yaml]: We don't need these

File: .coderabbit.yaml [hunk 10]
```diff
@@ -0,0 +1,280 @@
...
+
+  # Custom review instructions for specific file patterns
+  path_instructions:
+    # TypeScript/JavaScript files
+    - path: "**/*.{ts,tsx,js,jsx}"
+      instructions: |
+        NextJS:
+        - Ensure that "use client" is being used
+        - Ensure that only features that allow pure client-side rendering are used
+        - NextJS best practices (including file structure, API routes, and static generation methods) are used.
```
Comment (M4dhav) [inline .coderabbit.yaml]: We aren't going to use Next.js here, so this can be removed

File: .coderabbit.yaml [hunk 11]
```diff
@@ -0,0 +1,280 @@
...
+
+        TypeScript:
+        - Avoid 'any', use explicit types
+        - Prefer 'import type' for type imports
+        - Review for significant deviations from Google JavaScript style guide. Minor style issues are not a priority
+        - The code adheres to best practices associated with React
+        - The code adheres to best practices associated with React PWA
+        - The code adheres to best practices associated with SPA
+        - The code adheres to best practices recommended by lighthouse or similar tools for performance
```
Comment (M4dhav) [inline .coderabbit.yaml]: Irrelevant here

File: .coderabbit.yaml [hunk 12]
```diff
@@ -0,0 +1,280 @@
...
+        - The code adheres to best practices associated with React PWA
+        - The code adheres to best practices associated with SPA
+        - The code adheres to best practices recommended by lighthouse or similar tools for performance   
+        - The code adheres to best practices associated with Node.js
+        - The code adheres to best practices recommended for performance
+
+        Security:
+        - No exposed API keys or sensitive data
+        - Use expo-secure-store for sensitive storage
+        - Validate deep linking configurations
```
Comment (M4dhav) [inline .coderabbit.yaml]: Irrelevant here

File: .coderabbit.yaml [hunk 13]
```diff
@@ -0,0 +1,280 @@
...
+        Review the HTML code against the google html style guide and point out any mismatches. Ensure that:
+        - The code adheres to best practices recommended by lighthouse or similar tools for performance   
+
+    # CSS files
+    - path: "**/*.css"
+      instructions: |
+        Review the CSS code against the google css style guide and point out any mismatches. Ensure that: 
+        - The code adheres to best practices associated with CSS.
+        - The code adheres to best practices recommended by lighthouse or similar tools for performance.  
+        - The code adheres to similar naming conventions for classes, ids.
```
Comment (M4dhav) [inline .coderabbit.yaml]: Irrelevant here

File: .coderabbit.yaml [hunk 14]
```diff
@@ -0,0 +1,280 @@
...
+        - Confirm architectural consistency with existing contracts (no unintended storage layout changes unless clearly documented).
+        - Flag major feature additions or architectural changes that were implemented without prior design discussion (if applicable).
+        - Flag pull requests that mix unrelated changes or multiple concerns in a single submission.      
+        - Ensure security-sensitive logic changes are not introduced without adequate test coverage.      
+        - Review for common smart contract vulnerabilities, including but not limited to:
+            - Reentrancy
+            - Improper input validation
+            - Access control bypass
+            - Integer overflows/underflows (if using unchecked blocks)
+            - Front-running risks where applicable
```
Comment (M4dhav) [inline .coderabbit.yaml]: Irrelevant here

File: .coderabbit.yaml [hunk 15]
```diff
@@ -0,0 +1,280 @@
...
+
+
+    # Javascript/Typescript test files
+    - path: "**/*.test.{ts,tsx,js,jsx}"
+      instructions: |
+        Review test files for:
+        - Comprehensive coverage of component behavior
+        - Proper use of @testing-library/react-native
```
Comment (M4dhav) [inline .coderabbit.yaml]: Irrelevant here

File: .coderabbit.yaml [hunk 16]
```diff
@@ -0,0 +1,280 @@
...
+      instructions: |
+        Review test files for:
+        - Comprehensive coverage of contract behavior.
+        - Coverage of success paths, edge cases, and failure/revert scenarios.
+        - Proper validation of access control restrictions.
+        - Verification of event emissions where applicable.
+        - Explicit validation of state changes after each relevant function call.
+        - Adequate test updates whenever contract logic is modified.
+        - Deterministic behavior (tests should not rely on implicit execution order or shared mutable state).
+        - Clear and descriptive test names that reflect the intended behavior being validated.
```
Comment (M4dhav) [inline .coderabbit.yaml]: Irrelevant here

File: .coderabbit.yaml [hunk 17]
```diff
@@ -0,0 +1,280 @@
...
+
+
+    # Asset files (images, fonts, etc.)
+    - path: "assets/**/*"
```
Comment (M4dhav) [inline .coderabbit.yaml]: Our assets aren't inside a subfolder

**Review by M4dhav — CHANGES_REQUESTED** [review]

**Comment by Muneerali199** [comment]
sir @M4dhav , I Applied all requested changes to .coderabbit.yaml and cleaned up irrelevant configurations.
